metabrainz/__init__.py

```diff
@@ -10,21 +15,30 @@ def create_app(config_path=None):
         use_debug_toolbar=True,
     )
 
-    # Configuration files
-    app.config.from_pyfile(os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '..', 'default_config.py'
-    ))
-    app.config.from_pyfile(os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '..', 'consul_config.py'
-    ), silent=True)
-    app.config.from_pyfile(os.path.join(
-        os.path.dirname(os.path.realpath(__file__)),
-        '..', 'custom_config.py'
-    ), silent=True)
-    if config_path:
-        app.config.from_pyfile(config_path)
+    # Load configuration files
+    # If we're running under a docker deployment, only use the consul configuration.
+    if deploy_env:
+        print("Running in production!");
+        consul_config = os.path.join( os.path.dirname(os.path.realpath(__file__)), 
+            '..', 'consul_config.py')
+        while not os.path.exists(consul_config):
+            print("No configuration file generated yet. Sleeping...");
+            sleep(2)
+
+        app.config.from_pyfile(consul_config, silent=True)
```

Comment:
If you're going to wait for this file, there's probably no need for `silent=True` - removing this check will make the app explicitly fail if you get here and it doesn't exist, which is probably a good idea

Reply:
Yes, it will restart the service, but I am not sure of all of the semantics in place, since our current setup is unclear to me and not properly documented. But keeping silent=True and explicitly checking I can guarantee what is going to happen and that way I can guarantee that it is going to do the right thing. I am ok with improving this once the underlying issue has been solved. For now, there is little harm in doing it this way.